apache/superset
# feat: Enable new dataset creation flow

### SUMMARY

This PR enables the new create dataset flow in dataset view, the `+ -> Data` menu in the app header, chart creation, and database creation. On the create dataset page (`/dataset/add/`), user is brought to their previous page when the cancel button is clicked and the create chart view when the create dataset button is clicked.

### TESTING INSTRUCTIONS

- Check all areas in Superset where user can create a dataset, it should bring you to the new create dataset page: `/dataset/add`
  - All creation spots: `+ Dataset` button in dataset view, the `+ -> Data` menu in the app header, chart creation, and database creation.
  - There should no longer be an Add Dataset modal in any area of the app
- On the new create dataset page, ensure that:
  - Cancel button brings you to the previous page you were on
  - Create Dataset button brings you to the create chart view with the newly created dataset selected

### ADDITIONAL INFORMATION
